diff --git a/canaille/forms.py b/canaille/forms.py
index 97be45b..a2af0cc 100644
--- a/canaille/forms.py
+++ b/canaille/forms.py
@@ -216,6 +216,9 @@ def profile_form(write_field_names, readonly_field_names):
     form = wtforms.form.BaseForm(fields)
     for field in form:
         if field.name in readonly_field_names - write_field_names:
+            if not field.render_kw:
+                field.render_kw = {}
+
             field.render_kw["readonly"] = "true"
 
     return form
diff --git a/canaille/oidc/oauth.py b/canaille/oidc/oauth.py
index 4a0e76f..20f8b3d 100644
--- a/canaille/oidc/oauth.py
+++ b/canaille/oidc/oauth.py
@@ -290,6 +290,8 @@ class RevocationEndpoint(_RevocationEndpoint):
 
 
 class IntrospectionEndpoint(_IntrospectionEndpoint):
+    CLIENT_AUTH_METHODS = ["client_secret_basic", "client_secret_post"]
+
     def query_token(self, token, token_type_hint):
         return query_token(token, token_type_hint)
 
@@ -396,7 +398,6 @@ class CodeChallenge(_CodeChallenge):
     def get_authorization_code_challenge_method(self, authorization_code):
         return authorization_code.challenge_method
 
-
 authorization = AuthorizationServer()
 require_oauth = ResourceProtector()
